components/Forms/LogHoursPopupWindowForm.tsx

```diff
@@ -5,43 +5,63 @@ import {
   LargeFormInput,
   LogHoursForm,
 } from '@/styles/components/Forms/logHoursPopupWindowForm.styles';
-import React from 'react';
+import React, { useState } from 'react';
 import { useForm } from 'react-hook-form';
 import PopupWindow from '@/components/shared/PopupWindow';
 import WindowFlow from '@/components/shared/WindowFlow';
+import VolunteerSignedEvents from '../Volunteer/VolunteerSignedEvents';
+import { QueriedVolunteerEventData } from 'bookem-shared/src/types/database';
 
 const LogHoursPopupWindowForm = ({
   setShowPopup,
+  successMessage,
+  errorMessage,
 }: {
   setShowPopup: React.Dispatch<React.SetStateAction<boolean>>;
+  successMessage: (message: string) => void;
+  errorMessage: (message: string) => void;
 }) => {
   // get functions from react hook form
   const { register, handleSubmit } = useForm();
 
+  const [selectedEvent, setSelectedEvent] =
+    useState<QueriedVolunteerEventData>();
+
   // TODO: combine event and program into one so users only need to select event
   // that they signed up for (which is automatically under a program)
-  const pages = ['Event', 'Program', 'Numbers', 'Comments'];
+  const pages = ['Event', 'Numbers', 'Comments'];
 
   // handle form submission by parsing data and calling createVolunteerLog
   const onSubmit = (data: any) => {
     const results = JSON.stringify({
-      // event: data.Event,
+      eventId: selectedEvent?._id,
       hours: parseInt(data.NumberOfHours),
       date: data.DateOfVisit,
       feedback: data.Comment,
       numBooks: data.NumberOfBooks,
     });
-    console.log(results);
     createVolunteerLog(results);
   };
 
   // sends a post request to insert the volunteer form
   const createVolunteerLog = async (data: any) => {
-    // TODO: implement endpoint for VolunteerEventApplication and call it
-    await fetch('/api/volunteerLogs/create', {
-      method: 'POST',
-      body: data,
-    });
+    try {
+      // Implement endpoint for VolunteerEventApplication and call it
+      const response = await fetch('/api/volunteerLogs/create', {
+        method: 'POST',
+        body: data,
+      });
+      if (response.status === 200) {
+        const message = (await response.json()).message;
+        setShowPopup(false);
+        successMessage(message);
+      } else {
+        const message = (await response.json()).message;
+        errorMessage(message);
+      }
+    } catch (err) {
+      errorMessage('Sorry an error occurred');
+    }
```

Comment:
When we get the response from backend, we need to decide whether it's a success or a failure.